cloudpickle/cloudpickle.py

```diff
@@ -154,7 +154,13 @@ def _whichmodule(obj, name):
     # Protect the iteration by using a list copy of sys.modules against dynamic
     # modules that trigger imports of other modules upon calls to getattr.
     for module_name, module in list(sys.modules.items()):
-        if module_name == '__main__' or module is None:
+        # Some modules such as coverage can inject non-module objects inside
+        # sys.modules
+        if (
+                module_name == '__main__' or
+                module is None or
+                not isinstance(module, types.ModuleType)
```

Comment:
maybe we can do a `issubclass(type(module), types.ModuleType)` check instead? WDYT?

Reply:
`isinstance(module, types.ModuleType)` is simpler and cleaner.